postgres/bin/dump-to-s3.sh

```
@@ -1,19 +1,29 @@
 #!/bin/bash
 # Usage:  dump-to-s3.sh <s3://...dump> [dbname]
-# Expects a DATABASE_URL environment variable that is the DB to dump
-# Optionally, a dbname can be supplied as the second argument
-# to override the name from the DATABASE_URL
+# Expects a DATABASE_URL environment variable that provides the DB connection details.
+# Optionally, a dbname can be supplied as the second argument to override the name from the DATABASE_URL.
+
 set -euf -o pipefail
 
 cleanup() { rv=$?; if [ -f /tmp/db.dump ]; then shred -u /tmp/db.dump; fi; exit $rv; }
 trap cleanup EXIT
 
-NAME=${2:-$NAME}
-CONNECT_DB_URL="postgres://$USER@$HOST:$PORT/$NAME"
+# Extract database name from DATABASE_URL if not provided as an argument
+DBNAME=${2:-$(echo "$DATABASE_URL" | sed -E 's|.*/([^/?]+).*|\1|')}
+CONNECT_DB_URL="${DATABASE_URL%/*}/$DBNAME"
+
+# Get PostgreSQL server version
+SERVER_VERSION=$(psql "$CONNECT_DB_URL" -tAc "SHOW server_version;" | cut -d '.' -f 1)
+PG_DUMP="pg_dump-$SERVER_VERSION"
+
+if ! command -v "$PG_DUMP" &>/dev/null; then
+  echo "Error: pg_dump for version $SERVER_VERSION is not installed." >&2
+  exit 1
```

rather than failing hard here, let's make this a warning and default to the latest version

---

Done.